pkg/api/update/update.go

```diff
@@ -43,12 +44,15 @@ func (handle *Handler) Handle(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	select {
-	case chanValue := <-lock:
-		defer func() { lock <- chanValue }()
-		handle.fn()
-	default:
-		log.Debug("Skipped. Another update already running.")
+	var images []string
+	if r.URL.Query().Has("image") {
+		images = strings.Split(r.URL.Query().Get("image"), ",")
+	} else {
+		images = nil
 	}
 
+	chanValue := <-lock
+	defer func() { lock <- chanValue }()
+	handle.fn(images)
```

Comment:
This still need to check if the lock is taken before running the update, otherwise it's kind of pointless to have a lock...
```suggestion
select {
case chanValue := <-lock:
	defer func() { lock <- chanValue }()
	handle.fn(images)
default:
	log.Debug("Skipped. Another update already running.")
}
```

Reply:
I could be wrong but I understood the original code to skip an update altogether when one is already running, while my change would make it wait until it's done before proceeding with the new update. Although I didn't actually look this up.

My idea was that you can issue multiple update requests in short succession, for different images (as a busy CI system might do), and they won't interfere with each other.

Reply:
@piksel so my change makes the lock useless because it proceeds regardless of whether or not the lock was taken, but we should still allow multiple update requests for different images to be queued, no? I don't know how to do that, though, but I could find out of course.

Reply:
If the channel is sized to 1, it will block until the lock can be acquired. It does make more sense to queue up the requests now when they might have different parameters. I think it's generally fine, but since it's changed behaviour, maybe we can check if `len(images) > 0` and only block if that is the case. This is just to prevent any previously working setups from suddenly start checking continuously.

Reply:
If the channel is sized to 1, it will block until the lock can be acquired. It does make more sense to queue up the requests now when they might have different parameters. I think it's generally fine, but since it's changed behaviour, maybe we can check if `len(images) > 0` and only block if that is the case. This is just to prevent any previously working setups from suddenly start checking continuously.

Reply:
(I get that this was what your original code was doing, I just misunderstood what you meant with "useless" above. In my suggestion it should've been followed by a `return`)